tests/lib/tools/tests.cleanup

```diff
@@ -0,0 +1,79 @@
+#!/bin/sh -e
+
+show_help() {
+    echo "usage: tests.cleanup prepare"
+    echo "       tests.cleanup defer <cmd> [args]"
+    echo "       tests.cleanup restore"
+}
+
+cmd_prepare() {
+    if [ -e defer.sh ]; then
+        echo "tests.cleanup: cannot prepare, already prepared" >&2
+        exit 1
+    fi
+    truncate --size=0 defer.sh
+    chmod +x defer.sh
+}
+
+cmd_defer() {
+    if [ ! -e defer.sh ]; then
+        echo "tests.cleanup: cannot defer, must call tests.prepare first" >&2
+        exit 1
+    fi
+    echo "$*" >> defer.sh
+}
+
+cmd_restore() {
```

Comment:
Could this be called on restore-each?

Reply:
See my answer above. It perhaps might but until we strongly agree that cleanup has implicit prepare/restore and that the semantics are just right, I would not go there yet.